echo_server.py

```diff
@@ -13,24 +22,38 @@ def create_server_socket():
 
 
 def receive_msg(server_socket, buffsize):
+    """
+    Receive a string from a client.
+
+    Wait until a client connects, then keep receiving the message in chunks
+    equal to buffsize until the message is over, and return the connection,
+    address and message.
+    """
     conn, addr = server_socket.accept()
+
     message = ''
     keep_going = True
     while keep_going:
         pkt = conn.recv(buffsize)
-        print repr(pkt)
+        sys.stdout.write('server: ' + repr(pkt))
         message = '{}{}'.format(message, pkt)
 
         # For a last packet of buffsize, this loop iterates one more time
         if len(pkt) < buffsize:
             keep_going = False
-            print keep_going
+            sys.stdout.write(str(keep_going))
+
     return conn, addr, message
 
 
 def parse_request(header):
+    """
+    Return the uri.
+    """
     header = header.split('\r\n')
-    first_line = header[0].split()
+    first_line = header[0]
+
+    first_line = first_line.split(' ')
```

Comment:
Don't split on a string containing a space.  If things are separated by more than one space you'll get odd results.  If you mean "split on whitespace" just don't provide an argument to `str.split()`